src/Lifecycle.js

```diff
@@ -646,13 +654,15 @@ export function stopMatrixClient(unsetClient=true) {
     ActiveWidgetStore.stop();
     IntegrationManagers.sharedInstance().stopWatching();
     if (DMRoomMap.shared()) DMRoomMap.shared().stop();
+    EventIndexPeg.stop();
     const cli = MatrixClientPeg.get();
     if (cli) {
         cli.stopClient();
         cli.removeAllListeners();
 
         if (unsetClient) {
             MatrixClientPeg.unset();
+            EventIndexPeg.unset().done();
```

Comment:
`done` is on it's way out with removal of `bluebird`, so let's avoid adding new usages of it. Probably just `unset` is enough...?